src/App.tsx

```diff
@@ -1,204 +1,104 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useState, useEffect, useCallback, useMemo } from 'react';
+import * as todosService from './api/todos';
+
+import { ErrorMessages } from './components/ErrorsMessage';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+
+import { Todo } from './types/Todo';
+import { Field } from './types/Field';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosCounter, setTodosCounter] = useState<number>(0);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [field, setField] = useState<Field>(Field.ALL);
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    const timerId = window.setTimeout(() => setErrorMessage(''), 4000);
+
+    return () => clearTimeout(timerId);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    const todosActive = todos.filter(todo => !todo.completed);
+
+    setTodosCounter(todosActive.length);
+  }, [todos]);
+
+  const changeComplete = useCallback(
+    (todoChanged?: Todo) => {
+      const changedTodos = todos.map(todo => {
+        if (!todoChanged) {
+          return { ...todo, completed: !todo.completed };
+        }
+
+        if (todo.id !== todoChanged.id) {
+          return todo;
+        } else {
+          return { ...todo, completed: !todo.completed };
+        }
+      });
+
+      setTodos(changedTodos);
+    },
+    [todos],
+  );
+
+  const filteredTodos = useMemo(() => {
+    const preparedTodos = [...todos];
+
+    switch (field) {
+      case Field.COMPLETED:
+        return preparedTodos.filter(todo => todo.completed);
+      case Field.ACTIVE:
+        return preparedTodos.filter(todo => !todo.completed);
+      default:
+        return preparedTodos;
+    }
+  }, [todos, field]);
 
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          {/* this button should have `active` class only if all todos are completed */}
-          <button
-            type="button"
-            className="todoapp__toggle-all active"
-            data-cy="ToggleAllButton"
+        <Header
+          todos={todos}
+          setTodos={setTodos}
+          todosCounter={todosCounter}
+          changeComplete={changeComplete}
+        />
+
+        {todos && (
```

Comment:
```suggestion
        {!!todos.length && (
```

`todos` is always `true` because it's an array